root/static/scripts/guess-case/modes.js

```diff
@@ -278,7 +278,9 @@ exports.French = assign({}, DefaultMode, {
     return DefaultMode.runPostProcess(is)
       .replace(/([!\?;:]+)/gi, ' $1')
       .replace(/([«]+)/gi, '$1 ')
-      .replace(/([»]+)/gi, ' $1');
+      .replace(/([»]+)/gi, ' $1')
+      .replace(/^(Le\s|La\s|Les\s|L[’'])(\S+)$/gi,
+               (_, m1, m2) => m1 + capitalize(m2));
```

Comment:
In this specific case, it makes sense to replace ASCII apostrophe with typographical one.
(It won’t be the first time that case guessing function is used for typographical changes.)